src/main/webapp/js/map-loader.js

```diff
@@ -1,29 +1,180 @@
+
+  /* Initialize map with three fixed markers*/
+  function createDefaultMap(){
+    map = new google.maps.Map(document.getElementById('map'), {
+      center: {lat: 1.34, lng: 103.8},
+      zoom: 11
+    });
+  }
+
+  exports.createMapForAboutUsPage = function(){
+    createDefaultMap();
+    createMarker(1.280547, 103.844502, 'Maxwell Food Centre', '1 Kadayanallur St 069184');
+    createMarker(1.308398, 103.885808, 'Old Airport Road Food Centre', '51 Old Airport Rd, Singapore 390051');
+    createMarker(1.331814, 103.938867, 'Bedok 85 Market', '85 Bedok North Street 4, Singapore 460085');
+  };
+
+  exports.createMapForUserPage = function(){
+    createDefaultMap();
+    map.setZoom(16);
+    getCurrentLoc();
+  };
+
+  exports.searchLocationByText = function() {
+    var request = {
+        location: currentPos,
+        query: document.getElementById("location_input").value,
+        radius: "100"
+      };
+    service = new google.maps.places.PlacesService(map);
+    service.textSearch(request, function(results, status) {
+
+      if (status === google.maps.places.PlacesServiceStatus.OK) {
+        clearExistingMarkers();
+        if(results.length == 0){
+          alert("No result. Please check your input.");
+        }
+        else{
+          for (var i = 0; i < results.length; i++) {
+            createMarkerWithSubmit(results[i]);
+          }
+        }
+      }
+      else{
+        alert("Sorry. Search not available for now:(");
+      }
+    });
+    map.setCenter(currentPos);
+  };
+
+  /* Clear exiting marker if there are markers left from previous search. */
+  function clearExistingMarkers(){
+    for(var i =0; i < editMarkers.length; i++){
+      editMarkers[i].setMap(null);
+    }
+    editMarkers = [];
+  }
+
+  /** Creates a marker that shows a textbox the user can confirm the location and submit. */
+  function createMarkerWithSubmit(place){
+    const position = place.geometry.location;
+    var editMarker = new google.maps.Marker({
+      position: {lat: position.lat(), lng: position.lng()},
+      map: map
+    });
+    editMarkers.push(editMarker);
+    const infoWindow = new google.maps.InfoWindow({
+      content: place.name
+    });
+    infoWindow.open(map, editMarker);
+
+    editMarker.addListener('click', () => {
+      infoWindow.setContent(buildInfoWindowWithSubmit(place));
+      infoWindow.open(map, editMarker);
+    });
+  }
+
+  /** Builds and returns HTML elements that show an editable textbox and a submit button. */
+  function buildInfoWindowWithSubmit(place){
+    const position = place.geometry.location;
+
+    const para = document.createElement("P");
+    const text = document.createTextNode(place.name);
+    para.appendChild(text);
+    const button = document.createElement('button');
+    button.appendChild(document.createTextNode('Add this location'));
+    button.onclick = () => {
+      addLocationInfoToForm(position.lat(), position.lng(), place.name, place.place_id);
```

Comment:
@BenMQ  Here's the logic of populating values for hidden inputs. When the button "Add this location" is hit, this function "addLocationInfoToForm" creates hidden inputs, which are meant to be submitted to servlet along with the form later. Here place.place_id is passed in as location_id, it is an id maintained by google map api, which uniquely identifies each location on google map. The search result "place" contains this information so I just directly take it.

Reply:
good to know, thanks for explaining!